### PR TITLE
Force read-mode in `av.open`

### PR DESCRIPTION
**Pull request by ClaytonJY**
The `av.open` functions checks input metadata to determine the mode to open with ("r" or "w"). If an input to `decode_audio` is found to be in write-mode, it will fail. Forcing read mode solves this.

I encountered this in a FastAPI application, where I used the `UploadFile` type to accept a file in my API endpoint. This uses a `SpooledTemporaryFile` underneath. In some circumstances, such as with small input files, this spooled file will have mode "w+b" when we first attempt to read it, causing `decode_audio` to try opening with `av.open` in write mode, and failing. Forcing to read-mode fixes this.

It would not make sense for `av.open` in `decode_audio` to ever use write-mode, so this should not negatively impact any other uses.

### CODE REVIEW

**Review by threeal — APPROVED** [review]

**Review by MahmoudAshraf97 — APPROVED** [review]

**Comment by gonzaloplaza** [comment]
Hi there, we're experiencing the same issue. Please do you know if this kind of fixes can be released soon? Many thanks 🙏🏻

**Comment by ClaytonJY** [comment]
@gonzaloplaza while I have some approvals it looks like we're waiting on an "approving review"; this is my first PR here so I don't know how many folks can provide that or what to expect on timeline :shrug:

**Review by nguyendc-systran — APPROVED** [review]

**Pull request merged**
